example/big_seventh.cpp

```diff
@@ -0,0 +1,181 @@
+#include <limits>
+
+/*` So now we can demonstrate with some trivial calculations:
+*/
+
+//] //[big_seventh_example_1]
+
+int main()
+{
+
+//[big_seventh_example_2
+/*`Using a `typedef` like `cpp_bin_float_50` hides the complexity of multiprecision, and
+allows us to define variables with 50 decimal digit precision just like __fundamental `double`.
+*/
+  using boost::multiprecision::cpp_bin_float_50;
+
+  cpp_bin_float_50 seventh = cpp_bin_float_50(1) / 7;  // 1 / 7
+
+/*`By default, output would only show the standard 6 decimal digits,
+ so set precision to show all 50 significant digits, including any trailing zeros (to show the full implied 50 digit precision).
+*/
+  std::cout.precision(std::numeric_limits<cpp_bin_float_50>::digits10); // Show 50 decimal digit precision.
+  std::cout << std::showpoint << std::endl; // Append any trailing zeros.
+  std::cout << seventh << std::endl;
+/*`which outputs:
+
+  0.14285714285714285714285714285714285714285714285714
+
+We can also use Boost.Math __math_constants like [pi],
+guaranteed to be initialized with the very last bit of precision for the floating-point type.
+*/
+   std::cout << "pi = " << boost::math::constants::pi<cpp_bin_float_50>() << std::endl;
+   cpp_bin_float_50 circumference = boost::math::constants::pi<cpp_bin_float_50>() * 2 * seventh;
+   std::cout << "c =  "<< circumference << std::endl;
+
+/*`which outputs
+
+  pi = 3.1415926535897932384626433832795028841971693993751
+
+  c =  0.89759790102565521098932668093700082405633411410717
+*/
+//]  [/big_seventh_example_2]
+
+//[big_seventh_example_3
+/*`So using `cpp_bin_float_50` looks like a simple 'drop-in' for the __fundamental_type like 'double',
+but beware of less-than-expected precision from construction or conversion from `double` or other lower precision types.
+This is a mistake that is very easy to make,
+and very difficult to detect because the difference in precision is only visible after about the 17th decimal digit.
+
+We can show this by constructing a fraction one seventh from `double`:
+*/
+   cpp_bin_float_50 seventh_0 = cpp_bin_float_50(1/7);  // Avoid the schoolboy-error `double d7 = 1 / 7;` giving zero!
+   std::cout << "seventh_0 = "  << seventh_0 << std::endl;
+   // seventh_double0 = 0.0000000000000000000000000000000000000000000000000
+
+   cpp_bin_float_50 seventh_double = cpp_bin_float_50(1./7);  // Construct from double! (0.14285714285714)
+   std::cout << "seventh_double = "  << seventh_double << std::endl; // Boost.Multiprecision post-school error!
+   // seventh_double = 0.14285714285714284921269268124888185411691665649414
+
+/*`Did you spot the probably-unintended difference?  After the 17th decimal digit, result is apparently random
+and not the expected recurring pattern 14285714285714...
+
+The 'random' digits after digit 17 are from the `cpp_bin_float_50` representation of the `double` value 0.14285714285714
+which is different from the 'better' `cpp_bin_float_50` representation of the fraction 1/7
```

Comment:
This is a bit clunky but I don't know how to re-phrase - the cpp_bin_float_50 represents the double precision value exactly, but of course all the extra bits are set to zero.  The apparently "random" digits in the decimal representation aren't random at all - they're an artefact of the base-2 to 10 conversion (ie we can never represent the binary value exactly in base 10, so the decimal digits go on and on forever).  Apologies for being a pedant!

Reply:
Is this any better (if even clunkier?)   

The 'apparently random' digits after digit 17 are from the `cpp_bin_float_50` representation 
of the `double` value 0.14285714285714 that is, in effect, converted 
from 0.1428571428571400000000000000000000000000000000000
which is different from the 'nearest representable' or 'best' `cpp_bin_float_50` representation of the fraction 1/7

PS "The term, then, is obviously a relative one: my pedantry is your scholarship, his reasonable accuracy, her irreducible minimum of education and someone else’s ignorance."―H. W. Fowler, Modern English Usage.